web/book/src/internals/modules.md

```diff
@@ -0,0 +1,95 @@
+# Modules
+
+PRQL needs modules that would:
+
+- allow importing declarations from other files,
+- have namespaces for things like `std`,
+- have hierarchical structure so we can represent files in directories,
+- have unambiguous module structure within a project,
+- be able to compile individual files that are a part of a project.
+
+## Basic definition
+
+Module is a namespace that contains declarations.
+
+Modules can be defined with mod keyword and a code block encased in curly
+braces:
+
+```
+mod my_module {
+    let bangers = (from tracks | take 10)
+}
+```
+
+## File loading
+
+If a module definition of `my_module` does not have a code block, it's contents
+are loaded from `./my_module.prql`.
+
+If the file does not exist, contents are loaded from file
+`./my_module/mod.prql`.
+
+When compilation is invoked compiler will search for the root of the module
+structure of the project:
+
+1. attempt to load `./project.prql`. If successful use it as the root,
```

Comment:
What do you see as the difference between a project and a module? If we don't need a distinction, we can avoid having one at the moment, and just use `mod` for both...

Reply:
My intention was for project to represent the root of the module tree.

This may not be needed if we would also have `super` (a reference to parent module), which would be nice to have in any case.

Reply:
Does there need to be a "roof of the module"?
Shouldn't **one** module **always** be just **one** file?